docs/full/README.md

```diff
@@ -0,0 +1,152 @@
+
+## Target
+
+A Target refers to a category or tier of control planes that the CLI can interact with. There are currently two supported targets : kuberbetes (or k8s) and mission-control (or tmc) which corresponds to Kubernetes cluster endpoint type and Tanzu Mission Control endpoint types. A context is associated with one of the supported targets. Plugins are not necessarily and can often be associated with one of the targets as well. The see plugins that apply to a particular Target and not any others, run `tanzu <target>'.
+
+Similarly, commands from plugins that are associated with a target are unambiguously invoked by prefixing the command group with the target, like so:
+
+```console
+tanzu mission-control data-protection ...
+or
+tanzu k8s management-cluster ...
+```
+
+Note that today, the CLI supports omitting the target for historical reasons, but such omission only applies command for the k8s target.  (So `tanzu management-cluster ...` variant of the above example is valid, but not `tanzu data-protection ...`). The CLI team is exploring making the 'assumed target' configurable.
+
+## Interaction between CLI and its plugins
+
+The Core CLI plays several roles in plug architecture:
+
+### Central, consistent point of interaction
+
+All commands provided by the CLI are invocable via the tanzu binary, which in turn dispatches the command to the appropriate plugin, capturing output and errors from the latter in the process to return back to the user.
+
+Commands producing meaningful output consistently provide alternative output such as JSON, YAML or tabular formats using the `-o {json|yaml}` flag.
+
+### Plugin discovery and lifecycle management
+
+The CLI is configured to use a default plugin repository. Through various commands like `tanzu plugin search`, `tanzu plugin install`, `tanzu plugin group install`, the CLI provides various means to discover, then securely install or update plugins to serve specific needs of the user.
+
+Certain context endpoints will require that a specific set of plugins be installed so as enable proper interaction with said endpoints. For these, establish a connection with these endpoints will lead to the discovery and automatic installation of the additional plugins. Plugins installed through this mean of discovery are referred to as "context-scoped" plugins. To learn more about the context-scoped plugins, please check [context-scoped plugin installation](../full/context-scoped-plugins.md).
+
+For an overview on some of these plugin lifecycle commands, see the [Quickstart Guide](../quickstart/quickstart.md)
+For more details on these commands, see the [command reference](../cli/commands/tanzu_plugin.md)
+
+### Context management
+
+The CLI maintains a list of Contexts and an active Context for each Target type. A command from a plugin for use with a particular Target type will always be able to access the Context information necessary to interacted with the endpoint associated with the Context.
```

Comment:
Yeah, this seems confusing to me as well. How about something like this?

```suggestion
The CLI maintains a list of Contexts and an active Context for each Target type. A plugin command with a particular Target type will always be able to access the active context information by using the APIs exposed by the `tanzu-plugin-runtime` library. This will allow plugins to interact with the endpoint associated with the Context.
```

Reply:
@marckhouzam can you please check this PR to see if it looks correct? https://github.com/vmware-tanzu/tanzu-cli/pull/178